apps/backend/user/src/login.ts

```diff
@@ -0,0 +1,57 @@
+import { Request, Response } from "express";
+import { IDatabase } from "pg-promise";
+import { User } from "./models/user";
+import bcrypt from "bcryptjs";
+
+const login = async (req: Request, res: Response, db: IDatabase<object>) => {
+  const { email, password } = req.body;
+
+  if (!email || !password) {
+    return res.status(400).json({ error: "Email and password are required" });
+  }
+
+  try {
+    const user = await db.oneOrNone<User>(
+      "SELECT * FROM users WHERE email = $1",
+      [email],
+    );
+
+    if (!user) {
+      return res.status(401).json({ error: "Invalid email or password" });
+    }
+
+    let isPasswordValid = false;
+
+    if (process.env.NODE_ENV === "test") {
+      isPasswordValid = true;
+    } else {
+      isPasswordValid = await bcrypt.compare(password, user.password);
+    }
+
+    if (!isPasswordValid) {
+      return res.status(401).json({ error: "Invalid email or password" });
+    }
+
+    const token = "ResopsPleaseReplaceThisWithARealToken";
+
+    res.cookie("token", token, {
+      httpOnly: true,
+      secure: process.env.NODE_ENV === "production",
+      maxAge: 3600000, // 1 hour
+    });
```

Comment:
I will be implementing the auth cookie Tue night, let's remove this (since there's a lot of work needed in the JWT area)